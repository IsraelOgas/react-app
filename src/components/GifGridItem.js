import React from 'react'

export const GifGridItem = ( { id, title, url } ) => {

    // console.log( id, title, url )
    return (
        <a className="card animate__animated animate__fadeIn" href={ url } target="_blank" rel="noreferrer">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </a>
    )
}
