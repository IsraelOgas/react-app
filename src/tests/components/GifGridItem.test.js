import React from 'react';
import { shallow } from 'enzyme'
import { GifGridItem } from "../../components/GifGridItem"

describe( 'Pruebas en en componente GifGridItem', () => {

    const title = 'Un titulo';
    const url = 'https://url_random/img.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> )

    test( 'debe de mostrar el componente correctamente ', () => {
        expect( wrapper ).toMatchSnapshot();
    } )

    test( 'debe de tener un parrafo con el title', () => {
        const p = wrapper.find( 'p' );
        expect( p.text().trim() ).toBe( title );
    } )

    test( 'debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find( 'img' );
        // console.log( img.prop('src') );
        expect( img.prop( 'src' ) ).toBe( url );
        expect( img.prop( 'alt' ) ).toBe( title );
    } )

    test( 'debe de tener animate__fadeIn', () => {
        const a = wrapper.find( 'a' );
        const className = a.prop( 'className' );

        expect( className.includes( 'animate__fadeIn' ) ).toBe( true );
    } )

} )
