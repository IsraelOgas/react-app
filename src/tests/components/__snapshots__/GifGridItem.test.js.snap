// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en en componente GifGridItem debe de mostrar el componente correctamente  1`] = `
<a
  className="card animate__animated animate__fadeIn"
  href="https://url_random/img.jpg"
  rel="noreferrer"
  target="_blank"
>
  <img
    alt="Un titulo"
    src="https://url_random/img.jpg"
  />
  <p>
    Un titulo
  </p>
</a>
`;
